code/modules/crafting/recipes.dm

```diff
@@ -221,6 +221,21 @@
 	..()
 	blacklist += subtypesof(/obj/item/gun/energy/laser)
 
+/datum/crafting_recipe/sparker
+	name = "SPRK-12 Pistol"
+	tools = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	result = list(/obj/item/gun/energy/sparker)
+	reqs = list(/obj/item/gun/energy/gun/mini = 1,
+				/obj/item/stack/cable_coil = 5,
+				/obj/item/weaponcrafting/gunkit/sparker = 1)
+	time = 10 SECONDS
+	category = CAT_WEAPONRY
+	subcategory = CAT_WEAPON
+
+/datum/crafting_recipe/sparker/New()
+	..()
+	blacklist += subtypesof(/obj/item/gun/energy/gun/mini)
```

Comment:
do you want to include the original tiny one too?
```suggestion
	blacklist += typesof(/obj/item/gun/energy/gun/mini)
```

Reply:
no otherwise you would be unnable to craft it. The purpose of the blacklist in general is so if we have a laser gun, you can't craft it with the subtypes / but the base gun is not in the blacklist